main-app/app/scripts/providers/model/game-model.js

```diff
@@ -1,124 +1,51 @@
-                            {playerNumber: this.currentPlayer, chosenSquare: square}, {withCredentials: true})
-                            .success(function (data, status, headers, config) {
-                                populateBoard(data.gameboard);
-
-                                if (me.currentPlayer === 1 && p2State === 'human') {
-                                    me.currentPlayer = 2;
-                                }
-                                else
-                                if (me.currentPlayer === 2 && p1State === 'human') {
-                                    me.currentPlayer = 1;
-                                }
-
-                                if (data.outcome === 'Win') {
-                                    alert("A winner is Player " + data.winner + "!");
-                                    me.gameActive = false;
-                                }
-                                else if (data.outcome === 'Draw') {
-                                    alert("It's a tie!");
-                                    me.gameActive = false;
-                                }
-                            })
-                            .error(function (data, status, headers, config) {
-                                alert(data);
-                            });
+                    if (proxy.currentPlayer === 1) {
+                        return proxy.p1Name;
+                    }
+                    else if(proxy.currentPlayer === 2) {
+                        return proxy.p2Name;
                     }
                 };
 
-                var populateBoard = function(boardData) {
-
-                    me.board = boardData;
+                this.startNewGame = function() {
 
-                    for(var i=0;i<boardData.length;i++) {
+                    return proxy.startNewGame();
+                };
 
-                        switch(boardData[i]) {
-                            case '0':
-                                document.getElementsByClassName("boardSquare")[i].classList.remove("nought");
-                                document.getElementsByClassName("boardSquare")[i].classList.remove("cross");
-                                break;
-                            case '1':
-                                document.getElementsByClassName("boardSquare")[i].classList.add("cross");
-                                break;
-                            case '2':
-                                document.getElementsByClassName("boardSquare")[i].classList.add("nought");
-                                break;
-                        }
-                    }
+                this.makeMove = function(square) {
 
+                    return proxy.makeMove(square);
                 };
 
                 this.clearBoard = function() {
 
-                    populateBoard("000000000");
+                    return proxy.clearBoard();
```

Comment:
From the context this isn't clear - Are you clearing the board - or starting a new game. 
The API does not have a clear method.